pkg/cmd/cli/cmd/set/triggers.go

```diff
@@ -0,0 +1,744 @@
+package set
+
+import (
+	"fmt"
+	"io"
+	"os"
+	"reflect"
+	"strings"
+	"text/tabwriter"
+
+	"github.com/golang/glog"
+	"github.com/spf13/cobra"
+
+	kapi "k8s.io/kubernetes/pkg/api"
+	"k8s.io/kubernetes/pkg/api/meta"
+	kcmdutil "k8s.io/kubernetes/pkg/kubectl/cmd/util"
+	"k8s.io/kubernetes/pkg/kubectl/resource"
+	"k8s.io/kubernetes/pkg/runtime"
+
+	buildapi "github.com/openshift/origin/pkg/build/api"
+	buildutil "github.com/openshift/origin/pkg/build/util"
+	cmdutil "github.com/openshift/origin/pkg/cmd/util"
+	"github.com/openshift/origin/pkg/cmd/util/clientcmd"
+	deployapi "github.com/openshift/origin/pkg/deploy/api"
+	"github.com/openshift/origin/pkg/generate/app"
+	imageapi "github.com/openshift/origin/pkg/image/api"
+	"k8s.io/kubernetes/pkg/util/sets"
+)
+
+const (
+	triggersLong = `
+Set or remove triggers for build configs and deployment configs
+
+All build configs and deployment configs may have a set of triggers that result in a new deployment
+or build being created. This command enables you to alter those triggers - making them automatic or
+manual, adding new entries, or changing existing entries.
```

Comment:
Mostly I think that's a bug in the build config api.  Deployment config
supports what we want - to express "not automatic, but not deleted".
Removing triggers is a rare operation.   The effect the user wants is "on"
/ "off".  Not "remove / add".  If we force them to use remove, they can't
reason about BCs and DCs the same.

On Wed, Feb 24, 2016 at 12:49 PM, Ben Parees <notifications@github.com>
wrote:

> In pkg/cmd/cli/cmd/set/triggers.go
> <https://github.com/openshift/origin/pull/7423#discussion_r53977181>:
>
> > +	buildutil "github.com/openshift/origin/pkg/build/util"
> > +	cmdutil "github.com/openshift/origin/pkg/cmd/util"
> > +	"github.com/openshift/origin/pkg/cmd/util/clientcmd"
> > +	deployapi "github.com/openshift/origin/pkg/deploy/api"
> > +	"github.com/openshift/origin/pkg/generate/app"
> > +	imageapi "github.com/openshift/origin/pkg/image/api"
> > +	"k8s.io/kubernetes/pkg/util/sets"
> > +)
> > +
> > +const (
> > +	triggersLong = `
> > +Set or remove triggers for build configs and deployment configs
> > +
> > +All build configs and deployment configs may have a set of triggers that result in a new deployment
> > +or build being created. This command enables you to alter those triggers - making them automatic or
> > +manual, adding new entries, or changing existing entries.
>
> yes, but in the cause of buildconfig you *are* destroying data when they
> specify --manual, so why not force the user to express that?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/openshift/origin/pull/7423/files#r53977181>.
>